gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/process/repository/synchronizer/accesspoint/AccessPointSynchronizer.java

```diff
@@ -0,0 +1,98 @@
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.atomic.AtomicLong;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+public class AccessPointSynchronizer implements RepositorySynchronizer {
+
+    private final AccessPointFetcher accessPointFetcher;
+    private final DeployerFactory deployerFactory;
+    private final ThreadPoolExecutor syncFetcherExecutor;
+    private final ThreadPoolExecutor syncDeployerExecutor;
+
+    @Override
+    public Completable synchronize(Long from, Long to, List<String> environments) {
+        AtomicLong launchTime = new AtomicLong();
+        return accessPointFetcher
+            .fetchLatest(from, to, environments)
+            .subscribeOn(Schedulers.from(syncFetcherExecutor))
+            .rebatchRequests(syncFetcherExecutor.getMaximumPoolSize())
+            .flatMap(accessPoints -> Flowable.fromStream(accessPoints.stream().map(this::prepareForDeployment)))
+            .compose(upstream -> {
+                AccessPointDeployer accessPointDeployer = deployerFactory.createAccessPointDeployer();
+                return upstream
+                    .parallel(syncDeployerExecutor.getMaximumPoolSize())
+                    .runOn(Schedulers.from(syncDeployerExecutor))
+                    .flatMap(deployable -> deploy(accessPointDeployer, deployable))
+                    .sequential();
+            })
+            .count()
+            .doOnSubscribe(disposable -> launchTime.set(Instant.now().toEpochMilli()))
+            .doOnSuccess(count -> {
+                String logMsg = String.format(
+                    "%s access points synchronized in %sms",
+                    count,
+                    (System.currentTimeMillis() - launchTime.get())
+                );
+                if (from == -1) {
+                    log.info(logMsg);
+                } else {
+                    log.debug(logMsg);
+                }
+            })
+            .ignoreElement();
+    }
+
+    private AccessPointDeployable prepareForDeployment(final AccessPoint accessPoint) {
+        return AccessPointDeployable.builder().accessPoint(accessPoint).syncAction(SyncAction.DEPLOY).build();
+    }
+
+    private static Flowable<AccessPointDeployable> deploy(
+        final AccessPointDeployer accessPointDeployer,
+        final AccessPointDeployable deployable
+    ) {
+        return accessPointDeployer
+            .deploy(deployable)
+            .andThen(accessPointDeployer.doAfterDeployment(deployable))
+            .andThen(Flowable.just(deployable))
+            .onErrorResumeNext(throwable -> {
+                log.error(throwable.getMessage(), throwable);
+                return Flowable.empty();
+            });
+    }
+
+    @Override
+    public int order() {
+        return 0;
```

Comment:
License synchronizer already uses this order. Could you redistribute the value occordingly to avoid duplicate?